frostsnap_core/src/device.rs

```diff
@@ -796,45 +804,63 @@ impl<S: NonceStreamSlot + core::fmt::Debug> FrostSigner<S> {
         );
 
         self.mutate(Mutation::NewKey {
-            key_id,
+            key_id: access_structure_ref.key_id,
             key_name,
             purpose,
         });
         self.mutate(Mutation::NewAccessStructure {
-            key_id,
-            access_structure_id,
+            access_structure_ref,
             threshold,
             kind: AccessStructureKind::Master,
         });
         self.mutate(Mutation::SaveShare(Box::new(SaveShareMutation {
-            key_id,
-            access_structure_id,
+            access_structure_ref,
             encrypted_secret_share,
         })));
     }
 
     pub fn tell_coordinator_about_backup_load_result(
         &mut self,
-        phase: LoadKnownBackupPhase,
+        phase: LoadBackupPhase,
         secret_share: SecretShare,
     ) -> impl IntoIterator<Item = DeviceSend> {
-        let share_index = phase.share_image.share_index;
-        let access_structure_ref = phase.access_structure_ref;
-        let ok = match phase.check_entered_backup(secret_share) {
-            Some(complete_share) => {
-                self.tmp_loaded_backups.push(complete_share);
-                true
+        let mut ret = vec![];
+
+        match phase {
+            LoadBackupPhase::Known(load_known_backup) => {
```

Comment:
The loading of known/unknown backups seems quite intertwined with this this design, but in my mind they are quite separate tasks except for the entering share on device stage. Will keep this in mind as I continue reviewing.

Reply:
Telling the user to enter a backup is the same between checking a backup and entering it for restoration. I didn't check if "checking" the backup still works while testing come to think of it...